core/src/main/scala/akka/kafka/KafkaConnectionCheckerSettings.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka
+
+import akka.actor.ActorSystem
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.FiniteDuration
+
+class KafkaConnectionCheckerSettings private[kafka] (val maxRetries: Int,
+                                                     val checkInterval: FiniteDuration,
+                                                     val factor: Double,
+                                                     val dispatcher: String) {
+
+  private def copy(maxRetries: Int = maxRetries,
+                   checkInterval: FiniteDuration = checkInterval,
+                   factor: Double = factor,
+                   dispatcher: String = dispatcher): KafkaConnectionCheckerSettings =
+    new KafkaConnectionCheckerSettings(
+      maxRetries,
+      checkInterval,
+      factor,
+      dispatcher
+    )
+
+  def withMaxRetries(maxRetries: Int): KafkaConnectionCheckerSettings =
+    copy(maxRetries = maxRetries)
+  def withFactor(factor: Double): KafkaConnectionCheckerSettings = copy(factor = factor)
+  def withDispatcher(dispatcher: String): KafkaConnectionCheckerSettings = copy(dispatcher = dispatcher)
+
+  /** Scala API */
+  def withCheckInterval(checkInterval: FiniteDuration): KafkaConnectionCheckerSettings =
+    copy(checkInterval = checkInterval)
+
+  /** Java API */
+  def withCheckInterval(checkInterval: java.time.Duration): KafkaConnectionCheckerSettings =
+    copy(checkInterval = checkInterval.asScala)
+
+  def getMaxRetries: Int = maxRetries
+  def getCheckInterval: java.time.Duration = checkInterval.asJava
+  def getFactor: Double = factor
```

Comment:
The Java getters are not required.

Reply:
remove them all or just 'getFactor' and 'getMaxRetries'?

Reply:
Remove all of them.